Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -0,0 +1,316 @@
+            assert(false)
+        }
+    }
+}
+
+// Private initialization code
+// The rule store can compile JSON rule files into a private format which is cached on disk.
+// On app boot, we need to check if the ruleStore's data is out-of-date, or if the names of the rule files
+// no longer match. Finally, any JSON rule files that aren't in the ruleStore need to be compiled and stored in the
+// ruleStore.
+@available(iOS 11, *)
+extension ContentBlockerHelper {
+    fileprivate func loadJsonFromBundle(forResource file: String, completion: @escaping (_ jsonString: String) -> Void) {
+        DispatchQueue.global().async {
+            guard let path = Bundle.main.path(forResource: file, ofType: "json"),
+                let source = try? NSString(contentsOfFile: path, encoding: String.Encoding.utf8.rawValue) as String else {
+                    return
+            }
+            DispatchQueue.main.async {
+                completion(source)
+            }
+        }
+    }
+
+    fileprivate func lastModifiedSince1970(path: String) -> TimeInterval? {
+        do {
+            let url = URL(fileURLWithPath: path)
+            let attr = try FileManager.default.attributesOfItem(atPath: url.path)
+            let date = attr[FileAttributeKey.modificationDate] as? Date
+            return date?.timeIntervalSince1970
+        } catch {
+            return nil
+        }
+    }
+
+    fileprivate func dateOfMostRecentBlockerFile() -> TimeInterval {
+        let blocklists = blocklistBasic + blocklistStrict
+        return blocklists.reduce(TimeInterval(0)) { result, filename in
+            guard let path = Bundle.main.path(forResource: filename, ofType: "json") else { return result }
+            let date = lastModifiedSince1970(path: path) ?? 0
+            return date > result ? date : result
+        }
+    }
+
+    fileprivate func removeAllRulesInStore(completion: @escaping () -> Void) {
+        guard let ruleStore = ruleStore else { return }
+
+        ruleStore.getAvailableContentRuleListIdentifiers { available in
+            guard let available = available else {
+                completion()
+                return
+            }
+            var completionCount = 0
+            available.forEach {
+                ruleStore.removeContentRuleList(forIdentifier: $0) { _ in
+                    completionCount += 1
+                    if completionCount == available.count {
+                        completion()
+                    }
+                }
+            }
+        }
+    }
+
+    // If any blocker files are newer than the date saved in prefs,
+    // remove all the content blockers and reload them.
+    // Pass true to completion block if the rule store was modified.
+    fileprivate func removeOldListsByDateFromStore(completion: @escaping (_ modified: Bool) -> Void) {
+        let fileDate = self.dateOfMostRecentBlockerFile()
```

Comment:
all calls on WKContentRuleListStore (i.e. ruleStore) are async, and this code calls into that to delete the stored rules